apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -1736,16 +1737,20 @@ header_to_endpoint_top(Header) ->
     CleanedH = aehttp_dispatch_ext:cleanup_genesis(HMap),
     maps:put(<<"hash">>, aec_base58c:encode(block_hash, Hash), CleanedH).
 
+block_to_endpoint_gossip_map(Block) ->
+    BMap = aec_blocks:serialize_to_map(Block),
+    aehttp_dispatch_ext:cleanup_genesis(BMap).
+
 block_to_endpoint_map(Block) ->
     block_to_endpoint_map(Block, #{tx_objects => false}).
 
 block_to_endpoint_map(Block, Options) ->
-    SerializeFun =
+    Encoding =
         case maps:get(tx_objects, Options, false) of
-            true -> fun aec_blocks:serialize_client_readable/1;
-            false -> fun aec_blocks:serialize_to_map/1
+            false -> message_pack; 
```

Comment:
This seems unintuitive to me. Why does `tx_objects` imply a certain encoding? Why not make it explicit instead?

Reply:
This is by client's API request. I agree on that making it more explicit would be better. I'll discuss it with @dadaphl if it would be a big change on his side.